DownFall/Models/Board.swift

```diff
@@ -174,27 +174,76 @@ class Board: Equatable {
         }
         return nil
     }
+    
+    private func use(_ ability: AnyAbility) {
+        if let playerCoord = self.tiles(of: .player(.zero)).first {
+            if case TileType.player(let data) = tiles[playerCoord].type {
+                let newData = data.use(ability)
+                var newTiles = tiles
+                newTiles[playerCoord.row][playerCoord.column] = Tile(type: .player(newData))
+                tiles = newTiles
+            }
+        }
```

Comment:
It sorta sucks that we have to know to use this before using each ability.  Is there a better way to do this?

Reply:
There doesnt seem like a better way, unless we handle all the item cases.  then we can put it at the top of the `private func use(_ ability: AnyAbility, on targets: [TileCoord], input: Input) -> Transformation` function